case/interactive/T0000_rackhd_OSInstall.py

```diff
@@ -100,8 +100,11 @@ def test(self):
                             self.log("WARNING", "There is {0} active workflow(s) on node {1}."
                                                 "Will not continue with OS installtion on this node\n".
                                         format(len(rsp), node_id))
-                            node_info = (node_id, node)
-                            node_os_install_list.remove(node_info)
+                            index = 0
+                            for node_id_pend, node_pend in node_os_install_list:
+                                if node_id_pend == node_id:
+                                    node_os_install_list.pop(index)
+                                index = index + 1
```

Comment:
This is a node traverse inside a node traverse, which is not good.
I believe what you want to do is to delete this key-value pair from the dict: `del(node_os_install_list[node_id])`, considering the probable key missing (which will not in this test), you can try...except to protect the code:

    try:
        del(node_os_install_list[node_id])
    except KeyError:
        pass

P.S the node_os_install_list is actually a dict, right?

Reply:
No, it's a list like [(node1_id, (node1_value1, node1_value2)),  (node2_id, (node2_value1, node2_value2)), ......] got from "self.monorail.get_nodes("compute").items()". Each time executing get_nodes function calling monorail restful for same node_id will get different tuple of node_value.
So here we can't just del by node_id like we tried before.

Reply:
The try exception sounds good, I can add that.

Reply:
Oh, seems we can't del a dic by key here, and using for loop, exception will never show here, so no need to add the try exception. And, the "node_id" pending on discovery can always be found in "node_os_install_list". :)

Reply:
@JuneZhou19 
1. you can use `self.monorail.get_nodes("compute")` without `items()` from the very beginning to manage maps in dict and simplify code, in line 71 and 72
2. in a dict, `del(some_dict[some_key])` is ok but same sentence is not available for list
3. I agree that try... except... is not needed in this context

Reply:
good idea! PR submitted just now.